kusto/cloudinfo_test.go

```diff
@@ -0,0 +1,93 @@
+package kusto
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetrieveCloudInfoMetadataSuccessMultipleUrls(t *testing.T) {
+	var tests = []struct {
+		name              string
+		responsePayload   string
+		expectedCloudInfo *cloudInfo
+	}{
+		{
+			name:            "test_login_endpoint_1",
+			responsePayload: `{"AzureAD": {"LoginEndpoint": "https://login.microsoftonline.com","LoginMfaRequired": false,"KustoClientAppId": "db662dc1-0cfe-4e1c-a843-19a68e65be58","KustoClientRedirectUri": "https://microsoft/kustoclient","KustoServiceResourceId": "https://kusto.dev.kusto.windows.net","FirstPartyAuthorityUrl": "https://login.microsoftonline.com/f8cdef31-a31e-4b4a-93e4-5f571e91255a"  },  "dSTS": {"CloudEndpointSuffix": "windows.net","DstsRealm": "realm://dsts.core.windows.net","DstsInstance": "prod-dsts.dsts.core.windows.net","KustoDnsHostName": "kusto.windows.net","ServiceName": "kusto"}}`,
+			expectedCloudInfo: &cloudInfo{
+				loginEndpoint:          "https://login.microsoftonline.com",
+				loginMfaRequired:       false,
+				kustoClientAppId:       "db662dc1-0cfe-4e1c-a843-19a68e65be58",
+				kustoClientRedirectUri: "https://microsoft/kustoclient",
+				kustoServiceResourceId: "https://kusto.dev.kusto.windows.net",
+				firstPartyAuthorityUrl: "https://login.microsoftonline.com/f8cdef31-a31e-4b4a-93e4-5f571e91255a",
+			},
+		},
+		{
+			name:            "test_login_endpoint_2",
+			responsePayload: `{"AzureAD": {"LoginEndpoint": "https://login2.microsoftonline.com","LoginMfaRequired": true,"KustoClientAppId": "db662dc1-0cfe-4e1c-a843-19a68e65bxxx","KustoClientRedirectUri": "https://microsoft/kustoclient","KustoServiceResourceId": "https://kusto.dev.kusto.windows.net","FirstPartyAuthorityUrl": "https://login.microsoftonline.com/f8cdef31-a31e-4b4a-93e4-5f571e912xxx"  },  "dSTS": {"CloudEndpointSuffix": "windows.net","DstsRealm": "realm://dsts.core.windows.net","DstsInstance": "prod-dsts.dsts.core.windows.net","KustoDnsHostName": "kusto.windows.net","ServiceName": "kusto"}}`,
+			expectedCloudInfo: &cloudInfo{
+				loginEndpoint:          "https://login2.microsoftonline.com",
+				loginMfaRequired:       true,
+				kustoClientAppId:       "db662dc1-0cfe-4e1c-a843-19a68e65bxxx",
+				kustoClientRedirectUri: "https://microsoft/kustoclient",
+				kustoServiceResourceId: "https://kusto.dev.kusto.windows.net",
+				firstPartyAuthorityUrl: "https://login.microsoftonline.com/f8cdef31-a31e-4b4a-93e4-5f571e912xxx",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
You can do this sub test runner, but not a lot of reason to do it here.  See comment on runServer() about how to combine all these tests into a single table driven test.